src/integration/java/io/pinecone/integration/controlPlane/pod/CollectionTest.java

```diff
@@ -120,10 +110,17 @@ public void testIndexFromCollectionHappyPath() throws InterruptedException {
 
     @Test
     public void testCreateIndexFromCollectionWithDiffMetric() throws InterruptedException {
-        // Note collection == 1 index
-        String metricForNewIndex = IndexMetric.DOTPRODUCT.toString();
+        // Use a different metric than the source index
+        String[] metrics = {IndexMetric.COSINE.toString(), IndexMetric.EUCLIDEAN.toString(), IndexMetric.DOTPRODUCT.toString()};
+        String targetMetric = IndexMetric.COSINE.toString();
+        for (String metric : metrics) {
+            if (!metric.equals(indexMetric)) {
+                targetMetric = metric;
+            }
+        }
+
```

Comment:
I still don't get  this loop. We're saying if the `metric` does not equal the `indexMetric`, update `targetMetric` to be `metric`? For me, there are too many similarly-named variables here. Can we maybe switch `metric` to just `m`, and/or switch `indexMetric` to be something like `sourceIndexMetric`?